### PR TITLE
Fix/co 376 invalid cert

### PR DESCRIPTION
**Pull request by frisonisland**
### Goal

The API VerifyCertKey is broken, as it always returns invalid.
This fix addresses some problems, such as:

- parsing only private keys with header BEGIN PRIVATE RSA KEY (e.g.: BEGIN PRIVATE KEY does not work)
- using remote manager to ssh in to the server itself to run zmcertmgr command

### What's changed
- The API works as a proxy, it writes some .key, .crt and _ca.crt files locally and then uses zmcertmgr to verify them, just like as before. 
- I've removed all checks on files as they do not make sense, since zmcertmgr will take care of them
- zmcertmgr is run through ProcessBuilder and the API waits for it to exit
- zmcertmgr output is checked against "error" string being present.
- Made repo changes to use ibiblio to resolve artifacts from maven, so we can transitive download dependecies
- Add ivy configuration to download maven artifacts using default conf
- Added test to check validity return value against some mocked process output

### Additional notes
Tested the API in a real server and behaves as it should.
Still missing a test where we have to provide a valid chain certificate from a valid authority and not self signed

### CODE REVIEW

**Comment by frisonisland** [comment]
**I have added and then removed log4j dependency** (it was the same as the mailbox). **It was needed to run some tests**, because when trying to load mailbox classes it complains that it cannot find log4j. This is mostly due to the dependencies manager system.
I was trying to run some tests with Junit, JaCoCo and everything, but **if you take a look at the build.xml is the same as the old mailbox** and I don't want to make the same work twice. Instead **I will propose to put this API back in the mailbox repo.**

**Review by Polpetta — APPROVED** [review]
LGTM

**Review by keshavbhatt — APPROVED** [review]

**Comment by keshavbhatt** [comment]
> I will propose to put this API back in the mailbox repo

Make sense, but we need to make sure of not breaking references of repo somewhere else if it is being used by other projects. If that comes out true, we will be referencing mailbox there instead of this. Otherwise I have my approval on to make it part of mailbox.

**Comment by keshavbhatt** [comment]
Another thing to note about this PR is that we included a version bump for a future release, which sounds unusual and potentially cause inconsistencies or toruble while producing the actual release. I recommend avoiding bumping the version in feature PRs. we can do it in a separate release commit.

**Comment by frisonisland** [comment]
> Another thing to note about this PR is that we included a version bump for a future release, which sounds unusual and potentially cause inconsistencies or toruble while producing the actual release. I recommend avoiding bumping the version in feature PRs. we can do it in a separate release commit.

I did it because on devel we have already the next version. Maybe I could have done it on devel.
But once we merge into main we do not have to merge back main into a RC.
We have to define when to bump anyway ;) I say on devel just before the merge on main

**Comment by frisonisland** [comment]
Requested review again, as I had to make some changes: I **fixed some formatting error**, because replacing spaces with newlines broke the certificate headers. Also **files were saved in the wrong tmp directory**.
Had to add some comments about usage, as **the API is extremely picky**, because:

- content provided by JSON API and SOAP is different.
- copy pasting the crt and key can lead to many errors.
In the end the API just proxies to zmcertmgr. Spaces are replaced with newlines (except in-headers).
**In general the API fails when adding spaces just before, in-between, or after the base64 content.** And I think it makes sense.
Adding spaces in between BEGIN and END sections works.

**Review by Polpetta — APPROVED** [review]
LGTM

**Review by keshavbhatt — APPROVED** [review]
LGTM

**Pull request merged**
